### PR TITLE
Add new entry in settings under FF and internals

### PR DESCRIPTION
**Pull request by cmonfortep**
Task/Issue URL: https://app.asana.com/0/488551667048375/1204184007142125/f 

### Description
Shows device sync state (on/off)
only visible for internal users
will access sync settings screen (not part of this PR)

### Steps to test this PR

_Feature 1_
- [x] install internal flavor app
- [x] go to settings
- [x] ensure sync setting appears with status off
- [x] Go to internal sync settings and create an account
- [x] go back to main settings screen
- [x] ensure syn settings appears with status on
- [ ]

### UI changes
| Before  | After |
| ------ | ----- |
!(Upload before screenshot)|(Upload after screenshot)|

### CODE REVIEW

File: sync/sync-api/src/main/java/com/duckduckgo/sync/api/DeviceSyncState.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+interface DeviceSyncState {
+    /**
+     * Checks if feature should be visible to users.
+     */
+    fun isFeatureEnabled(): Boolean
+
+    /**
+     * Checks if user is signed in on this device.
+     */
+    fun isDeviceSyncEnabled(): Boolean
```
Comment (malmstein) [inline sync/sync-api/src/main/java/com/duckduckgo/sync/api/DeviceSyncState.kt:29]: nit: naming. The description and the method name are not consistent. Either call it `isSignedIn` or change the description

File: sync/sync-impl/src/test/java/com/duckduckgo/sync/impl/AppDeviceSyncStateTest.kt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+internal class AppDeviceSyncStateTest {
+
+    private val appBuildConfig: AppBuildConfig = mock()
+    private val syncFeature: SyncFeature = mock()
+    private val syncRepository: SyncRepository = mock()
+
+    @Test
+    fun whenUserSignedInThenDeviceSyncEnabled() {
+        whenever(syncRepository.isSignedIn()).thenReturn(true)
+        val appDeviceSyncState = AppDeviceSyncState(appBuildConfig, syncFeature, syncRepository)
```
Comment (malmstein) [inline sync/sync-impl/src/test/java/com/duckduckgo/sync/impl/AppDeviceSyncStateTest.kt:37]: Move this to a "@Before" method. There's no need to repeat the same line 3 times.

File: sync/sync-impl/src/test/java/com/duckduckgo/sync/impl/AppDeviceSyncStateTest.kt [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        assertTrue(appDeviceSyncState.isDeviceSyncEnabled())
+    }
+
+    @Test
+    fun whenInternalBuildThenFeatureEnabled() {
+        whenever(appBuildConfig.flavor).thenReturn(BuildFlavor.INTERNAL)
+        givenFeatureFlag(enabled = false)
+        val appDeviceSyncState = AppDeviceSyncState(appBuildConfig, syncFeature, syncRepository)
```
Comment (malmstein) [inline sync/sync-impl/src/test/java/com/duckduckgo/sync/impl/AppDeviceSyncStateTest.kt:45]: ditto: initialisation

File: sync/sync-impl/src/test/java/com/duckduckgo/sync/impl/AppDeviceSyncStateTest.kt [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        assertTrue(appDeviceSyncState.isFeatureEnabled())
+    }
+
+    @Test
+    fun whenFeatureFlagEnabledThenFeatureEnabled() {
+        whenever(appBuildConfig.flavor).thenReturn(BuildFlavor.PLAY)
+        givenFeatureFlag(enabled = true)
+        val appDeviceSyncState = AppDeviceSyncState(appBuildConfig, syncFeature, syncRepository)
```
Comment (malmstein) [inline sync/sync-impl/src/test/java/com/duckduckgo/sync/impl/AppDeviceSyncStateTest.kt:53]: ditto: initialisation

File: sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/SyncRepository.kt [hunk 5]
```diff
@@ -33,6 +33,7 @@ import timber.log.Timber
 
 interface SyncRepository {
     fun createAccount(): Result<Boolean>
+    fun isSignedIn(): Boolean
```
Comment (malmstein) [inline sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/SyncRepository.kt:36]: nit:  documentation

**Review by malmstein — APPROVED** [review]
works as expected. I've added a comment on the naming of an interface, rest looks good.

File: sync/sync-api/src/main/java/com/duckduckgo/sync/api/DeviceSyncState.kt (hunk 1 above)
Comment (cmonfortep) [reply sync/sync-api/src/main/java/com/duckduckgo/sync/api/DeviceSyncState.kt:29]: sure thing. renamed to `isUserSignedInOnDevice`.

File: sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/SyncRepository.kt (hunk 5 above)
Comment (cmonfortep) [reply sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/SyncRepository.kt:36]: no need, this is still inside `impl` module.

**Pull request merged**